### ISSUE TITLE
Marshmallow requirement currently incompatible with Prefect

### ISSUE THREAD

**Issue by phoebebright** [fivetran/great_expectations#860]
Prefect requires::

    marshmallow >= 3.0.0b19, <= 3.2.1

Great expectations::

    marshmallow>=2.0,<3.0

On running tests with python 3.6 and ver 3.2.1 of marshmallow, I get one error which may be related:

======================================================================================================== FAILURES =========================================================================================================
_____________________________________________________________________________ TestSchemaDeserialization.test_deserialize_with_dump_only_param _____________________________________________________________________________
tests/test_deserialization.py:1323: in test_deserialize_with_dump_only_param
    assert "size" not in result
E   AssertionError: assert 'size' not in {'name': 'Mick', 'size': 12}
============================================================================================== 1 failed, 926 passed in 5.82s ==============================================================================================

**Comment by jcampbell** [comment]
Unfortunately, marshmallow changed their API fairly significantly between 2 and 3, and they dropped support for python 2 in the 3-series releases.

I think to be able to support Marshmallow 3, we would need to created two series of GE releases. We don't currently have plans for that, but I could see it as a possibility.

**Comment by phoebebright** [comment]
How big a job do you think it would be to create a python3 branch?

**Comment by jcampbell** [comment]
At this point, the only incompatibility would be marshmallow, which means the primary issue would be managing different releases. Realistically, the issue will be how much energy to put into separate maintenance going forward. I'd be open to creating a beta release series now that drops python2 support, especially given how close we are to py2 EOL.

**Comment by joeschmid** [comment]
We also have the same issue, i.e. we'd like to run GE from a Prefect Flow but currently aren't able to due to this. Would be happy to beta test a py3 version.

**Comment by jcampbell** [comment]
I expect we'll be able to tackle the updates to update python 3-only (and so marshmallow 3) in late February, but could potentially put a beta-only version sooner if there's a lot of interest.

**Comment by sloria** [comment]
marshmallow maintainer here 👋  . Just glancing at the code, it looks like the changes to migrate to marshmallow 3 will be pretty minimal.

You'll need to add `**kwargs` to the `post_load` method here: https://github.com/great-expectations/great_expectations/blob/c4d54c53d3fe13f0cdfb4fc7a53f3228ab8db335/great_expectations/datasource/generator/table_generator.py#L29

```python
   def make_asset_configuration(self, data, **kwargs): 
```

and remove the `.data` here: https://github.com/great-expectations/great_expectations/blob/c4d54c53d3fe13f0cdfb4fc7a53f3228ab8db335/great_expectations/datasource/generator/table_generator.py#L79

```python
asset_name: assetConfigurationSchema.load(asset_config) for 
```

**Comment by jcampbell** [comment]
Thank you @sloria ! Yes--I'm definitely keen to get this migration done, and know we're not gated on you so much as not wanting to fork the codebase yet but giving a few users still on legacy python a bit more leniency. At this point, I expect 0.9.0 will be our last release with python 2 support.

**Comment by cicdw** [comment]
It appears this was resolved with the release of Great Expectations 0.9.7!  🎉🎉

**Comment by jcampbell** [comment]
Yes!! Fanfare! :)
